api/v1alpha1/initialbuildpipelineselector_types.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// Defines boolean value in a string to have not set as well.
+// +kubebuilder:validation:Enum=true,false,True,False,1,0
+type BoolString string
+
+const (
+	Bool_true  BoolString = "true"
+	Bool_false BoolString = "false"
+	Bool_True  BoolString = "True"
+	Bool_False BoolString = "False"
+	Bool_1     BoolString = "1"
+	Bool_0     BoolString = "0"
+)
+
+func IsBoolStringTrue(arg BoolString) bool {
+	return arg == Bool_true || arg == Bool_True || arg == Bool_1
+}
+
+// WhenConditions defines requirements when specified build pipeline must be used.
+// All conditions are connected via AND, whereas cases within any condition connected via OR.
+// Example:
+//
+//	language: java
+//	projectType: spring,quarkus
+//	annotations:
+//	   builder: gradle,maven
+//
+// which means that language is 'java' AND (project type is 'spring' OR 'quarkus') AND
+// annotation 'builder' is present with value 'gradle' OR 'maven'.
+type WhenCondition struct {
+	// Defines component language to match, e.g. 'java'.
+	// The value to compare with is taken from devfile.metadata.language field.
+	// +kubebuilder:validation:Optional
+	Language string `json:"language,omitempty"`
+
+	// Defines type of project of the component to match, e.g. 'quarkus'.
+	// The value to compare with is taken from devfile.metadata.projectType field.
+	// +kubebuilder:validation:Optional
+	ProjectType string `json:"projectType,omitempty"`
+
+	// Defines if a Dockerfile should be present in the component, e.g. 'true'.
+	// The value to compare with is taken from devfile components of image type.
+	// +kubebuilder:validation:Optional
+	DockerfileRequired BoolString `json:"dockerfile,omitempty"`
+
+	// Defines names of component to match, e.g. 'my-component'.
```

Comment:
Should it be `name` rather than `names`?

Reply:
No, because the pipeline supports components with the listed **names**. Probably have to improve description.

Reply:
I was thinking about it. The thing is that in most cases there will be only one value (and I didn't want to force user to create one-item-list all the time), but for some extra cases we might need a list.
Another option is to use `ArrayOrString` but it still doesn't give that easiness in reading and writing (by the user).

Reply:
The decision is not final, I'm open to discussion